### DISCUSSION TITLE
How do Reach dApps work on Mobile?

### DISCUSSION THREAD

**Discussion by kcelestinomaria** [reach-sh/reach-lang#1204] [General]
Hello Reach Fam,

Are Reach Contracts compatible on Mobile phones without Walletconnect? Can someone implement a Reach dApp that interacts with a Mobile Wallet or App entirely on Mobile without having to use Walletconnect? If not so, are there any plans to make this possible? I think it would really improve user experience of Reach dApps aside from helping making some features on complex dApps available for users who only use Mobile Phones(Smartphones to be specific). Also, so far the Reach Frontend works for Web Apps only.

**Comment by jeapostrophe** [comment]
A Reach application can use whatever wallet you want or none at all.

You could ask the user for their secret key and then call `newAccountFromSecret` --- https://docs.reach.sh/frontend/#js_newAccountFromSecret

If there is another wallet technology you want to use, then you can call `setProvider` --- https://docs.reach.sh/frontend/#js_setProvider --- and interact with however you want.

To my knowledge, there is no good mobile wallet that is not based on WalletConnect, so there are no other options, but if there were, that's how you'd support them and you don't NEED us to do it for you

**Comment by kcelestinomaria** [comment]
I see, can Reach connectors, particularly the ``AccountFromMnemonic`` feature be enabled to read encrypted keys aside from a bare private key, this makes it safe on mobile, cause mostly it's about signing transactions. @jeapostrophe

**Reply by kcelestinomaria** [reply]
It will do a good part in making reach smart contracts more compatible with mobile apps @jeapostrophe , that's what we've noticed on our end.

**Reply by jeapostrophe** [reply]
Yup!
